ONSdigital/respondent-home-ui
# SUS058 Inform Respondents that RH will be unavailable during Planned Maintenance

# Motivation and Context
As a: A Respondent
I need: to be informed that I will be prevented from using the system when it is made unavailable for planned maintenance
So that: I am aware in advance of planned unavailability

[SUS058](https://digitaleq.atlassian.net/wiki/spaces/RASB/pages/961282070/SUS058+Inform+Respondents+that+RH+will+be+unavailable+Unavailable+during+Planned+Maintenance)

# What has changed
- added a new maintenance middleware which polls redis for messages
- improved the flash middleware to avoid duplication of messages
- added a docker-compose file with RH and redis
- added a script for posting maintenance messages to redis

# How to test?
1. `docker-compose build && docker-compose up -d`
2. `pipenv run python scripts/planned_maintenance "[my message]"`
e.g.: `pipenv run python scripts/planned_maintenance "very soon"`
3. Open http://0.0.0.0:9092/
4. Check maintenance message is present in an info panel.
5. `pipenv run python scripts/planned_maintenance --remove`
6. Refresh the page.
7. Make sure the maintenance message is no longer present.

NB:
- TTL expiry can be added with the `-ttl [N sconds]` flag.

# Screenshots

![image](https://user-images.githubusercontent.com/28348457/49012003-f80c0000-f16f-11e8-9402-25b1f7886668.png)


![image](https://user-images.githubusercontent.com/28348457/48850785-46e91c80-eda1-11e8-8357-b4cef36c0987.png)

# Links

- [ras-rm-docker-dev PR](https://github.com/ONSdigital/ras-rm-docker-dev/pull/93)
- [ras-deploy PR](https://github.com/ONSdigital/ras-deploy/pull/103)
